### PR TITLE
fix: RD-14191-SECRET_MASKING_EXACT_PATH-change-the-response-body-of-the-function

### PR DESCRIPTION
**Pull request by eugene-lumigo**

### CODE REVIEW

File: src/spans/awsSpan.ts [hunk 1]
```diff
@@ -51,6 +51,8 @@ export const NEO4J_SPAN = 'neo4j';
 export const ENRICHMENT_SPAN = 'enrichment';
 export const PRISMA_SPAN = 'prisma';
 
+const clone = require('rfdc')({ proto: true });
```
Comment (harelmo-lumigo) [inline src/spans/awsSpan.ts:54]: What does the `proto: true` mean?

File: src/spans/awsSpan.ts [hunk 2]
```diff
@@ -242,7 +244,7 @@ export const getEndFunctionSpan = (functionSpan: GenericSpan, handlerReturnValue
   const ended = new Date().getTime();
   let returnValue: any;
   try {
-    returnValue = payloadStringify(data);
+    returnValue = payloadStringify(clone(data));
```
Comment (harelmo-lumigo) [inline src/spans/awsSpan.ts:247]: Maybe we should the `clone` inside `payloadStringify`? if we're concerned about it recursively cloning every subset of the object, let's make it:
```ts
const payloadStringify = payload => originalPayloadStringify(clone(data))
```
that way we won't have potential issue in otherexisting of future use cases of `payloadStringify`. WDYT?

File: src/spans/awsSpan.test.js [hunk 3]
```diff
@@ -444,6 +445,35 @@ describe('awsSpan', () => {
...
+        string: 'value',
+        object: {
+          string: 'value',
+          object: {
+            string: 'value',
+          },
+        },
+      },
+    };
+    const expectedData = clone(handlerReturnValue.data);
```
Comment (harelmo-lumigo) [inline src/spans/awsSpan.test.js:465]: Why do we need the `clone` here? can't we just check that fields we focus on are not changed to `****`?

File: src/spans/awsSpan.ts (hunk 1 above)
Comment (eugene-lumigo) [reply src/spans/awsSpan.ts:54]: I'll revert it

Comment (harelmo-lumigo) [reply src/spans/awsSpan.ts:54]: why revert? sounds like a good thing :)

File: src/spans/awsSpan.ts (hunk 2 above)
Comment (eugene-lumigo) [reply src/spans/awsSpan.ts:247]: @harelmo-lumigo does make sense. However, in this approach, we will need to calculate the time impact. 
I propose to have it inside `payloadStringify` but only in the 
```
if (getSecretMaskingExactPath()) {}
```
section. As we know it is implementation `LUMIGO_SECRET_MASKING_EXACT_PATH`

Comment (harelmo-lumigo) [reply src/spans/awsSpan.ts:247]: For now, let's not change it and keep it that way.
I'd suggest to add a ticket for considering the clone at a single point in the process later, as a separate change we can benchmark easily

**Review by harelmo-lumigo — APPROVED** [review]

**Pull request merged**